File: UniqueNumberII.java
public int[] singleNum(int[] arr) {
        // Code here
        int xor = 0;
        for(int num : arr){
            xor ^= num;
        }
        
        int rightmostSetBit = (xor & (xor - 1)) ^ xor;
        int x = 0;
        int y = 0;
        for(int num : arr){
            if((num & rightmostSetBit) != 0){
                x ^= num;
            }
            else{
                y ^= num;
            }
        }
        
        int[] res = {x, y};
        Arrays.sort(res);
        return res;
    }

File: UniqueNumberII.cpp
vector<int> singleNum(vector<int>& arr) {
        // Code here.
        int value = 0;
        for (int num : arr) {
            value ^= num;
        }
    
        // Find the rightmost set bit
        int rightMostSetBit = (value & (value - 1)) ^ value;
    
        int x = 0, y = 0;
        for (int num : arr) {
            if ((num & rightMostSetBit) != 0) {
                x ^= num;
            } else {
                y ^= num;
            }
        }
    
        vector<int> res = {x, y};
        sort(res.begin(), res.end());
        return res;
    }
